packages/core/src/agent/Dispatcher.ts

```diff
@@ -80,9 +81,11 @@ class Dispatcher {
         senderKey: outboundMessage.senderKey,
         returnRoute: true,
       })
-    } else if (outboundMessage) {
+    } else if (outboundMessage && isOutboundDIDCommV1Message(outboundMessage)) {
       outboundMessage.sessionId = messageContext.sessionId
       await this.messageSender.sendMessage(messageContext.agentContext, outboundMessage)
+    } else if (outboundMessage && isOutboundDIDCommV2Message(outboundMessage)) {
+      await this.messageSender.sendMessage(messageContext.agentContext, outboundMessage)
```

Comment:
Sessions allow to keep a transport alive even after you have processed a message or responded to it. this is what allows you to open a websocket connect with return_route enabled and keep receiving messages as they arrive at the mediator.

It is also what allows you to respond in the http response with a new didcomm message instead of sending it to the endpoint from the did document. 

This is mostly useful for mobile devices connecting to a mediator, however it also often provides a quicker response time as you reply in the response of the http request instead of sending the message to the http endpont of the other party.

Reply:
We kept setting `sessionId` for DidComm V2 as well but we do not currently use/check `session` within `sendDIDCommV2Message` function. 
As I understood session is closely tied to the return route concept which seems to be not applied to DidComm V2 messaging.

Reply:
Return route for didcomm v2: https://github.com/decentralized-identity/didcomm-messaging/blob/main/extensions/return_route/main.md

How are you implementing edge mediation wihtout return routing?